src/components/Cluster/NewCluster/ReleaseSelector.js

```diff
@@ -23,22 +23,37 @@ const FlexRowDiv = styled.div`
   }
 `;
 
+const MarginlessReleaseComponentLabel = styled(ReleaseComponentLabel)`
+  margin: 0;
+`;
+
 class ReleaseSelector extends React.Component {
   state = {
     kubernetesVersion: '',
     loading: true,
   };
 
   componentDidMount() {
-    const kubernetesVersion = this.props.selectedRelease
-      ? this.props.releases[this.props.selectedRelease].components.find(
-          (component) => component.name === 'kubernetes'
-        ).version
-      : undefined;
+    this.setState({
+      kubernetesVersion: this.extractKubernetesVersion(),
+      loading: false,
+    });
+  }
 
-    this.setState({ kubernetesVersion, loading: false });
+  componentDidUpdate({ selectedRelease: prevRelease }) {
+    if (prevRelease !== this.props.selectedRelease) {
+      this.setState({ kubernetesVersion: this.extractKubernetesVersion() });
+    }
   }
 
+  extractKubernetesVersion = () => {
+    const { selectedRelease, releases } = this.props;
+
+    return releases[selectedRelease]?.components.find(
+      (component) => component.name === 'kubernetes'
+    ).version;
```

Comment:
Can the component named `kubernetes` not be there? Asking because `find()` will return `undefined` in that case, and trying to get the`version` property will throw an error

Reply:
I think it should always be there, but I'll add `?.` for good measure